### ISSUE TITLE
Date module stopped working on 3.5.0

### ISSUE THREAD

**Issue by ifliunt** [polybar/polybar#2273]
## Checklist

* [x] I have read the appropriate section in the [contributing
  guidelines](https://github.com/polybar/polybar/blob/master/CONTRIBUTING.md)
* [x] I believe this issue is a problem with polybar itself and not a misconfiguration on my part.
* [x] I have searched for other open and closed [issues](https://github.com/polybar/polybar/issues?q=is%3Aissue) that
  may have already reported this problem.
* [x] I have checked the [known issues](https://github.com/polybar/polybar/wiki/Known-Issues) page for this problem.

## Describe the bug

The date module stopped working after updating polybar to 3.5.0
**Expected behavior:**
![image](https://user-images.githubusercontent.com/61592878/101261894-0b7cd900-3743-11eb-8813-b561980d6989.png)

**Actual behavior:**
polybar fails to start.
output of `polybar top_bar`:
`Floating point exception (core dumped)`

**Was it working before?**
* Did you also experience this bug in an earlier version of polybar? - no
* If no, what was the last version where this worked correctly? - 3.4.3

## To Reproduce

```dosini
[module/date]
type = internal/date
interval = 0
date = %{A1:gsimplecal:}%a, %d %b %I:%M:%S %P%{A}


```

## Polybar Log

```
      Stack trace of thread 11535:
                                                 #0  0x00005603eb40fb70 _ZN7polybar7modules12timer_moduleINS0_11date_moduleEE6runnerEv (polybar + 0x1bab70)
                                                 #1  0x00007fe3159fcc24 execute_native_thread_routine (libstdc++.so.6 + 0xcfc24)
                                                 #2  0x00007fe3157b43e9 start_thread (libpthread.so.0 + 0x93e9)
                                                 #3  0x00007fe3156e2293 __clone (libc.so.6 + 0x100293)
                                                 
                                                 Stack trace of thread 11534:
                                                 #0  0x00007fe3156d15b9 __spawnix (libc.so.6 + 0xef5b9)
                                                 #1  0x00007fe3156d1d4b __spawni (libc.so.6 + 0xefd4b)
                                                 #2  0x00007fe3156d156f posix_spawn@@GLIBC_2.15 (libc.so.6 + 0xef56f)
                                                 #3  0x00007fe3156590e4 _IO_proc_open@@GLIBC_2.2.5 (libc.so.6 + 0x770e4)
                                                 #4  0x00007fe31565933f popen@@GLIBC_2.2.5 (libc.so.6 + 0x7733f)
                                                 #5  0x00005603eb4f4e74 _ZN5i3ipc14get_socketpathB5cxx11Ev (polybar + 0x29fe74)
                                                 #6  0x00005603eb4246a9 _ZN7polybar7modules9i3_module6updateEv (polybar + 0x1cf6a9)
                                                 #7  0x00005603eb42a39a _ZN7polybar7modules12event_moduleINS0_9i3_moduleEE6runnerEv (polybar + 0x1d539a)
                                                 #8  0x00007fe3159fcc24 execute_native_thread_routine (libstdc++.so.6 + 0xcfc24)
                                                 #9  0x00007fe3157b43e9 start_thread (libpthread.so.0 + 0x93e9)
                                                 #10 0x00007fe3156e2293 __clone (libc.so.6 + 0x100293)
                                                 
                                                 Stack trace of thread 11531:
                                                 #0  0x00007fe3157ba6a2 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0 + 0xf6a2)
                                                 #1  0x00007fe3159f6c11 __gthread_cond_wait (libstdc++.so.6 + 0xc9c11)
                                                 #2  0x00005603eb3cb1de n/a (polybar + 0x1761de)
                                                 #3  0x00007fe3159fcc24 execute_native_thread_routine (libstdc++.so.6 + 0xcfc24)
                                                 #4  0x00007fe3157b43e9 start_thread (libpthread.so.0 + 0x93e9)
                                                 #5  0x00007fe3156e2293 __clone (libc.so.6 + 0x100293)
                                                 
                                                 Stack trace of thread 11536:
                                                 #0  0x00007fe3156e2285 __clone (libc.so.6 + 0x100285)
                                                 #1  0x0000000000000000 n/a (n/a + 0x0)

```

## Environment:
* WM: i3
* Distro: ArchLinux
* Output of `polybar -vvv`:

```
polybar 3.5.0

Features: +alsa +curl +i3 +mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +randr (+monitors) +composite +xkb +xrm +xcursor

Build type: Release
Compiler: /usr/bin/c++
Compiler flags: -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -Wall -Wextra -Wpedantic -O3 -DNDEBUG
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now 
```

**Comment by patrick96** [comment]
Thanks for the report. Was able to reproduce this on my end as well

**Comment by patrick96** [comment]
The issue is `interval = 0`. Intervals should be strictly larger than 0, but the modules should enforce that without crashing the entire bar. Will add a patch to check that the interval is `> 0`.

**Comment by ifliunt** [comment]
> The issue is `interval = 0`. Intervals should be strictly larger than 0, but the modules should enforce that without crashing the entire bar. Will add a patch to check that the interval is `> 0`.

Thanks for a quick update.

**Comment by patrick96** [comment]
Fixed in #2274
